packages/calcite-components/src/components/accordion-item/accordion-item.tsx

```diff
@@ -243,12 +241,42 @@ export class AccordionItem implements ConditionalSlotComponent {
 
   @Listen("calciteInternalAccordionChange", { target: "body" })
   updateActiveItemOnChange(event: CustomEvent): void {
-    this.requestedAccordionItem = event.detail
-      .requestedAccordionItem as HTMLCalciteAccordionItemElement;
-    if (this.el.parentNode !== this.requestedAccordionItem.parentNode) {
+    const [accordion] = event.composedPath();
+    const parent = closestElementCrossShadowBoundary<HTMLCalciteAccordionElement>(
+      this.el,
+      "calcite-accordion"
+    );
+
+    if (accordion !== parent) {
+      return;
+    }
+
+    this.determineActiveItem(parent.selectionMode, event.detail.requestedAccordionItem);
+    event.stopPropagation();
+  }
+
+  @Listen("calciteInternalAccordionItemsSync", { target: "document" })
+  updateAccordionChildSync(event: CustomEvent): void {
+    const [accordion] = event.composedPath();
+    const accordionItem = this.el;
+
+    const willBeSyncedByDirectParent = accordionItem.parentElement === accordion;
```

Comment:
I think the name for this const is great but maybe a link to an issue or more info might be useful in a comment. What do you think?

Ideally, we can just do things one way and this will be cleaned up in the future.

Reply:
Good idea. I'll submit a follow-up PR for this to expedite this fix landing.